fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryWrapper.java

```diff
@@ -161,7 +162,8 @@ public IndexWriter getWriter(LuceneAnalyzerWrapper analyzerWrapper) throws IOExc
         if (writer == null || !writerAnalyzerId.equals(analyzerWrapper.getUniqueIdentifier())) {
             synchronized (this) {
                 if (writer == null || !writerAnalyzerId.equals(analyzerWrapper.getUniqueIdentifier())) {
-                    TieredMergePolicy tieredMergePolicy = new TieredMergePolicy()
+                    final IndexDeferredMaintenancePolicy deferredMergePolicy = state.store.getIndexDeferredMaintenancePolicy();
+                    TieredMergePolicy tieredMergePolicy = new FDBTieredMergePolicy(deferredMergePolicy.shouldAutoMergeDuringCommit())
```

Comment:
With this change, can we remove the other change that checks `getIndexDeferredMaintenancePolicy`? I _think_ that should be redundant, and having fewer places doing the same thing would make it easier to reason about. It would also mean that the existing tests we have would make sure that this code is working correctly to defer merges.

Reply:
✅